packages/frontend-shared/src/utils/addExternalLinkClickListener.ts

```diff
@@ -0,0 +1,17 @@
+export const addExternalLinkClickListener = (mutation: { executeMutation: ({ url: string }) => void }) => {
+  document.addEventListener('click', (event: MouseEvent) => {
```

Comment:
Do you think this function would be a good case for a composable (i.e. useExternalLinkClickListener)? If it was a composable we could take advantage of lifecycle functions to set up and clean up this event listener. Right now I don't think anything is cleaning up the event listener.